policy/policy-info-bigquery-repository.go

```diff
@@ -0,0 +1,134 @@
+	"bytes"
+	"cloud.google.com/go/civil"
+	"fmt"
+	"github.com/wopta/goworkspace/lib"
+	"github.com/wopta/goworkspace/models"
+	"regexp"
+	"strings"
+)
+
+type PolicyInfo struct {
+	Uid            string         `json:"uid" bigquery:"uid"`
+	ProductName    string         `json:"productName" bigquery:"productName"`
+	CodeCompany    string         `json:"codeCompany" bigquery:"codeCompany"`
+	ProposalNumber int            `json:"proposalNumber" bigquery:"proposalNumber"`
+	NameDesc       string         `json:"nameDesc" bigquery:"nameDesc"`
+	Status         string         `json:"status" bigquery:"status"`
+	Contractor     string         `json:"contractor" bigquery:"contractor"`
+	Price          float64        `json:"price" bigquery:"price"`
+	PriceMonthly   float64        `json:"priceMonthly" bigquery:"priceMonthly"`
+	Producer       string         `json:"producer" bigquery:"producer"`
+	ProducerCode   string         `json:"producerCode" bigquery:"producerCode"`
+	StartDate      civil.DateTime `json:"startDate" bigquery:"startDate"`
+	EndDate        civil.DateTime `json:"endDate" bigquery:"endDate"`
+	PaymentSplit   string         `json:"paymentSplit" bigquery:"paymentSplit"`
+}
+
+func getPoliciesInfoQueriesBigQuery(datasetID, tableID string, queries []models.Query, limit int) ([]PolicyInfo, error) {
+	query, params, err := buildPolicyInfoQuery(datasetID, tableID, queries, limit)
+	if err != nil {
+		return nil, err
+	}
+
+	policies, err := lib.QueryParametrizedRowsBigQuery[PolicyInfo](query, params)
+	if err != nil {
+		return nil, err
+	}
+
+	return policies, err
+
+}
+
+func buildPolicyInfoQuery(datasetID string, tableID string, queries []models.Query, limit int) (string, map[string]interface{}, error) {
+	var query bytes.Buffer
+	params := make(map[string]interface{})
+	fieldRegexp := regexp.MustCompile("^[a-zA-Z0-9.]*$")
+
+	query.WriteString(fmt.Sprintf("SELECT p.uid, p.name AS productName, p.codeCompany, CAST(p.proposalNumber AS INT64) AS proposalNumber, " +
+		"p.nameDesc,p.status, COALESCE(JSON_VALUE(p.data, '$.contractor.name'), '') || ' ' || " +
+		"COALESCE(JSON_VALUE(p.data, '$.contractor.surname'), '') AS contractor, " +
+		"p.priceGross AS price, p.priceGrossMonthly AS priceGross, nn.name AS producer, p.producerCode, p.startDate, " +
+		"p.endDate, p.paymentSplit " +
+		"FROM `wopta.policiesView` p " +
+		"INNER JOIN `wopta.networkNodesView` nn ON nn.uid = p.producerUid " +
+		"WHERE"))
+
+	for index, q := range queries {
+		if index != 0 {
+			query.WriteString(" AND")
+		}
+
+		valuesParameters := make([]string, 0)
+		if len(q.Values) > 0 {
+			for _, v := range q.Values {
+				valueParameter := randomString(12)
+				params[valueParameter] = v
+				valuesParameters = append(valuesParameters, valueParameter)
+			}
+		} else {
+			// value parameter should be a random string of only letters
+			valueParameter := randomString(12)
+			params[valueParameter] = q.Value
+			valuesParameters = append(valuesParameters, valueParameter)
+		}
```

Comment:
Can we refactor this? the only change between single and multiple values i the value to be used, but the structure is the same: generate key, add value to map, append key to keyList.